tests/test_generation_utils.py

```diff
@@ -156,46 +157,106 @@ def _get_encoder_outputs(model, input_ids, attention_mask, num_interleave=1):
+            input_ids,
+            attention_mask=attention_mask,
+            do_sample=False,
+            num_beams=1,
+            max_length=max_length,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            output_scores=output_scores,
+            **logits_process_kwargs,
+        )
+
+        if model.config.is_encoder_decoder:
+            encoder_outputs, input_ids, attention_mask = self._get_encoder_outputs(model, input_ids, attention_mask)
+            kwargs["encoder_outputs"] = encoder_outputs
+
+        with torch.no_grad():
+            output_greedy = model.greedy_search(
                 input_ids,
-                attention_mask=attention_mask,
-                do_sample=False,
-                num_beams=1,
                 max_length=max_length,
-                **logits_process_kwargs,
+                attention_mask=attention_mask,
+                logits_processor=logits_processor,
+                output_attentions=output_attentions,
+                output_hidden_states=output_hidden_states,
+                output_scores=output_scores,
+                **kwargs,
             )
+        return model, output_greedy, output_generate
 
+    def test_greedy_generate(self):
+        # check `generate()` and `greedy_search()` are equal
+        for model_class in self.all_generative_model_classes:
+            model, output_greedy, output_ids_generate = self._greedy_generate(model_class)
+            self.assertIsNone(output_greedy.logits)
             if model.config.is_encoder_decoder:
-                encoder_outputs, input_ids, attention_mask = self._get_encoder_outputs(
-                    model, input_ids, attention_mask
-                )
-                kwargs["encoder_outputs"] = encoder_outputs
+                self.assertTrue(isinstance(output_greedy, GreedySearchEncoderDecoderOutput))
+                self.assertTrue(isinstance(output_ids_generate, GreedySearchEncoderDecoderOutput))
+                for key in [
+                    "encoder_attentions",
+                    "decoder_attentions",
+                    "encoder_hidden_states",
+                    "decoder_hidden_states",
+                ]:
+                    self.assertIsNone(output_greedy.get(key))
+                    self.assertIsNone(output_ids_generate.get(key))
+            else:
+                self.assertTrue(isinstance(output_greedy, GreedySearchDecoderOnlyOutput))
+                self.assertTrue(isinstance(output_ids_generate, GreedySearchDecoderOnlyOutput))
+                for key in [
+                    "attentions",
+                    "hidden_states",
+                ]:
+                    self.assertIsNone(output_greedy.get(key))
+                    self.assertIsNone(output_ids_generate.get(key))
+
+            self.assertListEqual(output_ids_generate.sequences.tolist(), output_greedy.sequences.tolist())
+
+    def test_greedy_generate_output_scores(self):
```

Comment:
awesome that you added all the tests! I think for `generate()` we have to do an exception here since the tests are probably the most expensive ones. So far we had one `test_greedy_generate()` test. Can we only add one additional test that directly tests all outputs (scores, attentions, hidden_states) ? So for now only two tests?